### PR TITLE
Add appveyor configuration for Windows based unit testing

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

To ensure Windows compatibility, run unit tests on a Windows platform using appveyor.

### CODE REVIEW

File: .appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Config file for automatic testing at appveyor.com
+
+version: '{branch}-{build}'
+
+build: off
+
+cache:
+  - "%LOCALAPPDATA%\\pip\\Cache"
+
+environment:
```
Comment (aronasorman) [inline .appveyor.yml:10]: Do we really want to run all these environments? We might just be duplicating the work done by our normal travis test environment already. Maybe let's just take python 2.7 and 3.6.

File: .appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# Config file for automatic testing at appveyor.com
+
+version: '{branch}-{build}'
+
+build: off
```
Comment (aronasorman) [inline .appveyor.yml:5]: What does this flag mean?

File: .appveyor.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      PYTHON_ARCH: '64'
+
+install:
+  # Prepend newly installed Python to the PATH of this build (this cannot be
+  # done from inside a powershell script as it would require us to restart the
+  # parent CMD process).
+  - "SET PATH=%PYTHON_HOME%;%PYTHON_HOME%\\Scripts;%PATH%"
+  # Check that we have the expected version and architecture for Python
+  - "python --version"
+  - "python -c \"import struct; print(struct.calcsize('P') * 8)\""
```
Comment (aronasorman) [inline .appveyor.yml:80]: What does this line do?

File: .appveyor.yml (hunk 2 above)
Comment (rtibbles) [reply .appveyor.yml:5]: It means we don't want to make any build artifacts.

File: .appveyor.yml (hunk 1 above)
Comment (rtibbles) [reply .appveyor.yml:10]: Not necessarily, I was just being thorough!

File: .appveyor.yml (hunk 3 above)
Comment (rtibbles) [reply .appveyor.yml:80]: It prints out the architecture of the windows VM we are running on for a sanity check in test logs.

**Comment by rtibbles** [comment]
This is broken, and I've failed to fix it.

**Pull request closed without merging**